### PR TITLE
Updating docs

### PR DESCRIPTION
**Pull request by sgoldfed**

### CODE REVIEW

File: docs/Rollup_basics.md [hunk 1]
```diff
@@ -13,42 +13,45 @@ and lower cost than on-chain Ethereum contracts.
 Despite running off-chain, ArbChains are _trustless_, meaning that any
 party can force an ArbChain to execute correctly according to the code of its contracts.
 
-You can make as many ArbChains as you want.
-Typically, you'll want one ArbChain for each dapp you want to create.
+You can make as many ArbChains as you want. Each ArbChain provides the same interface as the Ethereum blockchain, and contracts can be deployed on the chain at anytime using the `CREATE` and `CREATE2` opcode.
```
Comment (edfelten) [inline docs/Rollup_basics.md:16]: Should say that you can also deploy contracts by sending transactions to address zero on the chain, just as you would on Ethereum.

File: docs/Rollup_basics.md [hunk 2]
```diff
@@ -13,42 +13,45 @@ and lower cost than on-chain Ethereum contracts.
...
+> Note: Earlier versions of Arbitrum did not fully support dynamic contract launching and required code for all contracts to be provided when initially launching the chain. We have now removed this limitation.
+
+Applications that reside on the same ArbChain can make synchronous calls to one another exactly as they would on Ethereum.
+Typically, you'll want to group dApps together into a single ArbChain if they need to interact and compose with one another. Applications that do not require interaction with others can be deployed on their own ArbChain.
```
Comment (edfelten) [inline docs/Rollup_basics.md:20]: Suggest saying that applications that do not require *synchronous* interaction will often be deployed on separate chains.

File: docs/Rollup_basics.md [hunk 3]
```diff
@@ -13,42 +13,45 @@ and lower cost than on-chain Ethereum contracts.
...
 
 Every ArbChain has a set of Validators, who monitor the chain to ensure that it executes correctly.
-Validators will deposit currency stakes, which they will lose if they behave dishonestly.
+Validators will deposit currency stakes, which they will lose if they behave dishonestly. Validating an Arbitrum Rollup chain is permissioneless; anyone who puts down a stake can serve as a validator.
```
Comment (edfelten) [inline docs/Rollup_basics.md:23]: Typo: "permissioneless"

File: docs/Rollup_basics.md [hunk 4]
```diff
@@ -13,42 +13,45 @@ and lower cost than on-chain Ethereum contracts.
...
 
 An ArbChain contains a set of contracts.
-Over time, you can start new contracts in an ArbChain, and contracts can self-destruct.
-There is one limitation, though: you have to provide the code for a contract when you initially build the chain.
-You can start as many instances of those contracts as you want,
-but you can't introduce new contract code into a chain after the chain has started.
-(This limitation will probably be lifted in future versions.)
+Over time, you can launch new contracts in an ArbChain, and contracts can self-destruct, exactly as they do on the Etherum blockchain.
+ArbChains accept EVM transactions that are byte-for-byte identical to Ethereum contracts.
```
Comment (edfelten) [inline docs/Rollup_basics.md:27]: identical to Ethereum *transactions*

File: docs/Rollup_basics.md [hunk 5]
```diff
@@ -13,42 +13,45 @@ and lower cost than on-chain Ethereum contracts.
...
 
-First, you compile your Solidity contracts, using the open-source Arbitrum compiler.
-This will bundle your contracts together, along with some runtime and "glue" code,
-into an Arbitrum executable file (or ".ao file").
-The .ao file is a single program, to be run on the Arbitrum Virtual Machine (AVM) architecture,
-which will manage and run the contracts and wallets needed for your chain.
-Think of it as an ArbChain in a box.
+Suppose you’re a developer who has written a dapp for the Ethereum platform. Arbitrum interoperates with Ethereum, so you can launch your dapp on an Arbitrum Rollup chain and get better speed and scalability. 
+
+You’re starting with a dapp—or you’re planning to develop one—that’s made up of some contracts written in Solidity, along with a browser-based front end.  Here’s how to use Arbitrum with your dapp. Arbitrum supports dynamic launching of contracts on deployed chains using Ethereum's CREATE and CREATE2 opcodes, so you can either launch a new Rollup chain for your dapp or deploy it to an existing one. Here, we describe the process for launching a new chain.
```
Comment (edfelten) [inline docs/Rollup_basics.md:35]: As above, can deploy contracts using CREATE and CREATE2, but also using transactions to address zero, just like Ethereum.  Standard build tools use the address zero tx method, so we should emphasize that more here.  Should also say that you can use standard build tools like Truffle (and Buidler?)

File: docs/Rollup_basics.md [hunk 6]
```diff
@@ -13,42 +13,45 @@ and lower cost than on-chain Ethereum contracts.
...
 
-Next, you start your ArbChain by making a call to an Ethereum contract called the EthBridge.
-The EthBridge will start an Ethereum contract to manage your chain,
+First, you’ll want to identify an initial set of validators for your chain. We’ll talk later about how you might choose validators, and why people might want to validate a chain. Of course, validators will be able to come and go at will once the chain is going.  
+
+When you’re ready to launch your chain, you send an Ethereum message to the EthBridge—the component that connects Arbitrum to Ethereum—telling it to launch your chain on Arbitrum Rollup, and identifying the contract’s initial validators.  The EthBridge will start an Ethereum contract to manage your chain,
```
Comment (edfelten) [inline docs/Rollup_basics.md:39]: Better to say:  send an Ethereum *transaction* to the EthBridge.

File: docs/Rollup_basics.md [hunk 7]
```diff
@@ -63,11 +66,14 @@ That's what makes ArbChains trustless.
...
 
-As a dapp developer, you would probably want to validate your dapp's ArbChain, to help ensure its correctness and show confidence in your dapp.
-Some users will want to validate as well, in order to protect their interest in the dapp's correct execution.
+Although validation is permissionless and anyone _can_ validate, it's important to make sure that each ArbChain always has one honest validator that is paying attention. As a dApp developer, you would probably want to validate your dApp's ArbChain, to help ensure its correctness and show confidence in your dApp.
+
+For developers who do not want to validate themselves, you'll be able to hire a validator-as-a-service to validate your ArbChain once Arbitrum is released for production.  
+You might want to hire more than one, if you or your users worry that one might misbehave. As long as any validator behaves honestly, the Arbitrum protocol guarantees that all contracts in the ArbChain will run correctly according to their code. 
+
+Some users will want to validate as well, in order to protect their interest in the dapp's correct execution. A healthy ArbChain with an engaged userbase will likely have a variety of validators.
```
Comment (edfelten) [inline docs/Rollup_basics.md:74]: After the "Some users will want" sentence, add one saying that users can hire validators if they want.

File: docs/Solidity_Support.md [hunk 8]
```diff
@@ -4,14 +4,21 @@ title: Solidity Support
 custom_edit_url: https://github.com/OffchainLabs/arbitrum/edit/master/docs/Solidity_Support.md
 ---
 
-Arbitrum Rollup allows you to deploy a set of Solidity contracts as on a trustless layer 2 sidechain. Arbitrum supports almost all solidity code as expected with a few exceptions.
+# TODO: We talk about Solidity, but should we change to EVM?
```
Comment (edfelten) [inline docs/Solidity_Support.md:7]: Yes, I think we want to say we support EVM, therefore Solidity support comes for free.

File: docs/Withdrawals.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+id: Withdrawals
+title: Withdrawing funds from Arbitrum
+sidebar_label: Withdrawals
+---
+
+# Withdrawing funds from Arbitrum
+
+As discussed in the section on [finality](Finality.md), because Arbitrum's execution happens optimistically, the Ethereum blockchain cannot immediately confirm the correct state and must wait for the challenge window to expire (or until all challenges are resolved). 
+
+When it comes to execution within the Rollup this does not pose any problem or add any delay. The [Arbitrum Rollup protocol](Rollup_Protocol.md) manages a tree of assertions, and allows validators to pipeline execution by continuing to build the tree even before all nodes are confirmed. This means that an honest validator can continue to advance the state of the machine with confidence (and the ability to enforce) that eventually Ethereum will recognize the honest branch as the correct and valid one. And although it will take some time for Ethereum to recognize which branch is correct, anyone that is validating the chain will immediately know this.
```
Comment (edfelten) [inline docs/Withdrawals.md:11]: At end of this paragraph, suggest saying that anyone that is validating the chain will immediately know which branch is correct and therefore which branch will eventually be accepted by the protocol.

File: docs/Withdrawals.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+The one part of the protocol that is affected by the confirmation delay is withdrawals. Since Arbitrum cannot undo a withdrawal once it has released funds from the L2, the system cannot allow funds to be withdrawn from the ArbChain until it has been confirmed on-chain that the withdrawal is valid.
```
Comment (edfelten) [inline docs/Withdrawals.md:13]: "confirmed on-chain" should be "confirmed on the Ethereum chain"

**Comment by sgoldfed** [comment]
PTAL @edfelten

**Pull request merged**